diff --git a/quiz.module b/quiz.module
index f9cf90e..13b1a01 100644
--- a/quiz.module
+++ b/quiz.module
@@ -2712,20 +2722,21 @@ function quiz_availability($quiz) {
     return t('This quiz only allows %num_attempts attempts. Anonymous users can only access quizzes that allows an unlimited number of attempts.',
       array('%num_attempts' => $quiz->takes));
   }
-
   $user_is_admin = user_access('edit any quiz content') || (user_access('edit own quiz content') && $quiz->uid == $user->uid);
   if ($user_is_admin || $quiz->quiz_always == 1) {
-    return TRUE;
+    $result = TRUE;
   }
 
   // Compare current GMT time to the open and close dates (which should still be
   // in GMT time).
-  $now = gmmktime();
+  $now = time();
 
-  if ($now >= $quiz->quiz_close || $now < $quiz->quiz_open) {
-    return t('This quiz is closed');
+  if (($now >= $quiz->quiz_close || $now < $quiz->quiz_open) && $quiz->quiz_always != 1) {
+    $result = t('This quiz is closed');
   }
-  return TRUE;
+  // Allow other modules to check availability
+  drupal_alter('quiz_availability_check', $result, $quiz);
+  return $result;
 }
